pkg/providers/snow/snow.go

```diff
@@ -103,27 +103,44 @@ func (p *SnowProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster
 }
 
 func CAPIObjects(ctx context.Context, clusterSpec *cluster.Spec, kubeClient kubernetes.Client) (controlPlaneSpec, workersSpec []byte, err error) {
-	controlPlaneObjs, err := ControlPlaneObjects(ctx, clusterSpec, kubeClient)
+	controlPlaneSpec, err = ControlPlaneSpec(ctx, clusterSpec, kubeClient)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	controlPlaneSpec, err = templater.ObjectsToYaml(controlPlaneObjs...)
+	workersSpec, err = WorkersSpec(ctx, clusterSpec, kubeClient)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	return controlPlaneSpec, workersSpec, nil
+}
+
+func ControlPlaneSpec(ctx context.Context, clusterSpec *cluster.Spec, kubeClient kubernetes.Client) (controlPlaneSpec []byte, err error) {
```

Comment:
I don't think we need this, we can reconcile `unstructured.Unstructured` directly